File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        l, r = 0, len(matrix) - 1
        mid = (l + r) // 2

        while l <= r:
            mid = (l + r) // 2
            if target not in matrix[mid]:
                if target < matrix[mid][0]:
                    r = mid - 1
                elif target > matrix[mid][-1]:
                    l = mid + 1
                else:
                    return False
            else: 
                return True
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] answer = {0, 0};
        for (int i = 0; i < nums.length; i++) {
            int index = target - nums[i];
                for (int j = i; j < nums.length; j++) {
                    if (nums[j] == index && i != j) {
                        answer[0] = i; 
                        answer[1] = j; 
                    }
                }
            }
            return answer;
        }
    }
